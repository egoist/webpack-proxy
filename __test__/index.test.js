const Config = require('webpack-chain')
const proxy = require('../lib')

test('chain', () => {
  const config = new Config()
  const rule = config.module.rule('js').test(/\.js$/)
  proxy.chain(rule, {
    loader: 'babel-loader',
    options: {
      babelrc: false
    }
  }, {
    version: '^1.0.0',
    dependencies: ['lol@^2.0.1']
  })
  expect(rule.uses.has('ensure-loader')).toBe(true)
  expect(rule.uses.has('proxy-loader')).toBe(true)

  for (const name in rule.uses.entries()) {
    expect(rule.use(name).get('options')).toMatchSnapshot(name)
  }
})
