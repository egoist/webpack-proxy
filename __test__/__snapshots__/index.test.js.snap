// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chain: ensure-loader 1`] = `
Object {
  "loader": "babel-loader",
  "version": undefined,
}
`;

exports[`chain: proxy-loader 1`] = `
Object {
  "loader": "babel-loader",
  "options": Object {
    "babelrc": false,
  },
}
`;
